src/components/text-field.js

```diff
@@ -38,89 +56,228 @@ const _textFieldClasses = ({ container, inputSize, success, error }) => ({
+    onChange(e.target.value)
   }
 
-  const handleFocus = () => {
-    setFocus(true)
+  const handleFocus = (i) => {
+    setFocus(i ? i : true)
     setFilled(false)
+    if (!isEdited) {
+      setIsEdited(true)
+    }
   }
 
   const handleBlur = () => {
     setFocus(false)
     if (value) setFilled(true)
   }
 
-  return (
-    <div className={textFieldClasses.container}>
-      {label && <div className='flex flex-row'>
-        <p className={textFieldClasses.label}>{label}</p>
-        {required && <span className='flex flex-row ml-5px text-error-500'>*</span>}
-      </div>}
-      <form onSubmit={(e) => {
-        e.preventDefault()
-        onSubmit({ ...e, target: e.target.children[0].children[0] })}
-      }>
-        <InputBase
-          {...inputProps}
-          classes={inputBaseClasses}
-          onFocus={handleFocus}
-          onBlur={handleBlur}
-          onChange={handleChange}
-          onClick={onClick}
-          onDelete={onDelete}
-          size={size}
-          deleteButton={!disabled && deleteButton}
-          required={required}
-          {...rest}
-        />
-      </form>
-      <div className="grid grid-cols-2">
-        {helperText && <p className={textFieldClasses.helperText}>{helperText}</p>}
-        {maxLength && <p className={textFieldClasses.wordCount}>{value.length || 0}/{maxLength}</p>}
+  const handleKeyDown = (e, i, inputID) => {
+    const BACKSPACE = 8
+    const LEFT_ARROW = 37
+    const RIGHT_ARROW = 39
+
+    if ([BACKSPACE, LEFT_ARROW].includes(e.keyCode)) {
+      if (e.keyCode === BACKSPACE) {
+        if (e.target.value) {
+          const v = [...linkedVals]
+          v.splice(i, 1, '')
+          setLinkedVals(v)
+          return onChange(v.join(''))
+        }
+      }
+      const prev = document.getElementById(`linked-${inputID}-${i}`)
+      if (prev) {
+        prev.focus()
+        setFocus(i)
+      }
+    }
+
+    if (e.keyCode === RIGHT_ARROW) {
+      const next = document.getElementById(`linked-${inputID}-${i+2}`)
+      if (next) {
+        next.focus()
+        setFocus(i+2)
+      }
+    }
+  }
+
+  const handleLinkedSubmit = (e) => {
+    e.preventDefault()
+    if (onSubmit) {
+      if (linkedVals.filter((r) => r).length !== linkedFields) {
+        setLinkedIncompleteError(true)
+      } else {
+        const target = document.getElementById(`linked-${inputID}-${focus}`)
+        target.blur()
+        setFocus(null)
+        setLinkedIncompleteError(false)
+        return onSubmit(linkedVals.join(''))
+      }
+    }
+  }
+
+  if (variant === 'linked') {
+    return (
+      <div className={textFieldClasses.container.linked.outer}>
+        {label && renderLabel({ label, required, textFieldClasses })}
+        <div className={textFieldClasses.container.linked.inner}>
+          {(new Array(linkedFields)).fill().map((_, i) => {
+            return (
+              <InputBase
+                {...inputProps}
+                {...rest}
+                key={i}
```

Comment:
Seems like using the index as a key is causing duplicated input value display when entered non sequentially.

Reply:
how do you mean?
enforced single character value for each input in recent commits, could you try and see if the same issue still occurs?

Reply:
When I assigned unique ID to the keys, the glitch was no longer showing. I was suspecting it is related to [this issue.](https://stackoverflow.com/questions/59517962/react-using-index-as-key-for-items-in-the-list) It is not showing with the latest commit though!

Reply:
ah i see, ya its either this, or before each input hasn't been restricted to single character, and having 1+ characters in each field messes up the states 🤦‍♀️